allennlp/commands/train.py

```diff
@@ -268,11 +270,48 @@ def train_model(
             include_package=include_package,
             world_size=1,
         )
-
         archive_model(serialization_dir, files_to_archive=params.files_to_archive)
         return model
     else:
-        raise NotImplementedError
+        device_id = parse_cuda_device(cuda_device)
+
+        if not isinstance(device_id, list):
+            raise ConfigurationError(
+                "Multiple cuda devices need to be configured to run distributed training."
+            )
+
+        logging.info("Switching to distributed training mode since multiple GPUs are configured")
+        os.environ["MASTER_ADDR"] = "127.0.0.1"
+        os.environ["MASTER_PORT"] = "29500"
```

Comment:
The reason I did not have this configurable was the assumption that this will only apply to single-node multi-GPU training. These properties will only have significance in a multi-node setup and I did not have such a setup to test till last week. But I was able to run distributed training a multi-node setup with only a few more changes. Hence this would be changed to be configurable.